ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/Server.java

```diff
@@ -251,30 +255,57 @@ private static HttpServerOptions createHttpServerOptions(final KsqlRestConfig ks
         .setPerFrameWebSocketCompressionSupported(true);
 
     if (tls) {
-      options.setUseAlpn(true).setSsl(true);
-
-      final String keyStorePath = ksqlRestConfig
-          .getString(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG);
-      final Password keyStorePassword = ksqlRestConfig
-          .getPassword(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG);
-      if (keyStorePath != null && !keyStorePath.isEmpty()) {
-        options.setKeyStoreOptions(
-            new JksOptions().setPath(keyStorePath).setPassword(keyStorePassword.value()));
+      if (isInternalListener) {
+        final String internalAlias = ksqlRestConfig
+            .getString(KsqlRestConfig.KSQL_SSL_KEYSTORE_ALIAS_INTERNAL_CONFIG);
+        setTlsOptions(ksqlRestConfig, options, internalAlias,
+            ksqlRestConfig.getClientAuthInternal());
+      } else {
+        final String externalAlias = ksqlRestConfig
+            .getString(KsqlRestConfig.KSQL_SSL_KEYSTORE_ALIAS_EXTERNAL_CONFIG);
+        setTlsOptions(ksqlRestConfig, options, externalAlias, ksqlRestConfig.getClientAuth());
```

Comment:
Just a suggestion but:

```suggestion
      final String ksConfigName = isInternalListener:
           ? KsqlRestConfig.KSQL_SSL_KEYSTORE_ALIAS_INTERNAL_CONFIG
           : KsqlRestConfig.KSQL_SSL_KEYSTORE_ALIAS_EXTERNAL_CONFIG;
           
      final String alias = ksqlRestConfig.getString(ksConfigName);
      setTlsOptions(ksqlRestConfig, options, alias, ksqlRestConfig.getClientAuthInternal());
```